src/util/walkMap.js

```diff
@@ -0,0 +1,29 @@
+const walkArray = (array, callback, path, target = []) => {
```

Comment:
I think we need some description of what this file does, I'm guessing it generates some type of tree.

Reply:
added comments